R/assay.R

```diff
@@ -1478,6 +1478,46 @@ tail.Assay <- function(x, n = 10L, ...) {
 # S4 methods
 #%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+setAs(
+  from = 'Assay5',
+  to = 'Assay',
+  def = function(from) {
+    data.list <- c()
+    for (i in c('counts', 'data', 'scale.data')) {
+      if (length(Layers(object = from, search = i)) > 1) {
+          from <- JoinLayers(object = from, 
+                             layers = i, 
+                             new = i) 
+      }
+      adata <- LayerData(object = from, layer = i)
+      if(inherits(x = adata, what = "IterableMatrix")) {
+        warning("Converting IterableMatrix to sparse dgCMatrix")
+        adata <- as(object = adata, Class = "dgCMatrix")
+      }
+      data.list[[i]] <- adata 
+    }
+    if (IsMatrixEmpty(x = data.list[["data"]])){
+      data.list[["data"]] <- data.list[["counts"]]
+    }
```

Comment:
Is it possible that both counts and data are empty?

Reply:
Yes, and just have `scale.data` layer. Counts will return empty. But Layers(object, "data") returns scale.data if data doesn't exist. So you'll get something like this: 

![Screen Shot 2023-02-02 at 9 57 28 AM](https://user-images.githubusercontent.com/59940281/216359231-17c6cefe-e739-4861-8eb7-2c305a0ad649.png)

So 2 things: 
- Does it make sense to return "scale.data" when you search for "data" layer?
- If you make data and counts empty it leads to issues with new() that we'd have to figure out.